aptos-move/aptos-vm/src/move_vm_ext/session/user_transaction_sessions/user.rs

```diff
@@ -102,8 +102,19 @@ impl<'r> UserSession<'r> {
 
         let init_func_name = ident_str!("init_module");
         for module in modules {
-            // Check if module existed previously. If not, we do not run initialization.
-            if module_storage.check_module_exists(module.self_addr(), module.self_name())? {
+            // INVARIANT:
+            //   We have charged for the old version (if it exists) before when pre-processing the
+            //   module bundle.
```

Comment:
What is this meant with this? Frankly I'm not quite understanding this old/new module logic. Is my understanding wrong that:

- For existing modules I have been charged in the past when I published them.
- Every module published now (whether upgrade or new) will be charged.
- We may also want to charge for compatibility check if an upgrade is performed

Reply:
The comment above only mentions the old version case (but they should already be charged - and is checked - for both cases).
 
"This covers compatibility": do you mean because it's charged twice both as old and new? That seems important anyway to account for module size changing right